src/main/java/com/icthh/xm/ms/configuration/config/SshTransportConfigCallback.java

```diff
@@ -1,50 +1,87 @@
 package com.icthh.xm.ms.configuration.config;
 
+import static org.apache.sshd.common.auth.UserAuthMethodFactory.PUBLIC_KEY;
+
 import com.icthh.xm.ms.configuration.config.ApplicationProperties.GitProperties.SshProperties;
-import com.jcraft.jsch.JSch;
-import com.jcraft.jsch.JSchException;
-import com.jcraft.jsch.Session;
+import jakarta.annotation.PreDestroy;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyPair;
+import java.security.PublicKey;
 import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
 import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.sshd.common.util.security.SecurityUtils;
 import org.eclipse.jgit.api.TransportConfigCallback;
+import org.eclipse.jgit.transport.CredentialsProvider;
 import org.eclipse.jgit.transport.SshSessionFactory;
 import org.eclipse.jgit.transport.SshTransport;
 import org.eclipse.jgit.transport.Transport;
-import org.eclipse.jgit.transport.ssh.jsch.JschConfigSessionFactory;
-import org.eclipse.jgit.transport.ssh.jsch.OpenSshConfig;
-import org.eclipse.jgit.util.FS;
+import org.eclipse.jgit.transport.sshd.JGitKeyCache;
+import org.eclipse.jgit.transport.sshd.ServerKeyDatabase;
+import org.eclipse.jgit.transport.sshd.SshdSessionFactoryBuilder;
 
 @Slf4j
 @RequiredArgsConstructor
 public class SshTransportConfigCallback implements TransportConfigCallback {
 
+    private static final String SSH_TEMP_DIR = "ssh-temp-dir";
     private final SshProperties sshProperties;
+    private final Path temporaryDirectory = getTempDirectory();
+
+    @SneakyThrows
+    private static Path getTempDirectory() {
+        return Files.createTempDirectory(SSH_TEMP_DIR);
+    }
+
+    @PreDestroy
+    public void destroy() {
+        try {
+            Files.deleteIfExists(temporaryDirectory);
```

Comment:
method will throw DirectoryNotEmptyException is directory is not empty. For what purpouse this directory will be used?